Remove remaining aconst_init, withfield, vnew refs

theresa-m opened the pull request:
Part 1: https://github.com/eclipse-openj9/openj9/pull/19771

Fixes: https://github.com/eclipse-openj9/openj9/issues/18316

hzongaro commented on runtime/compiler/ilgen/J9ByteCodeIterator.cpp:
```diff
@@ -611,8 +607,8 @@ const uint8_t TR_J9ByteCodeIterator::_byteCodeFlags[] =
                              0x01, // J9BCasyncCheck --- TODO: Is this the right size?
-                             0x03, // J9BCaconst_init
-                             0x03, // J9BCwithfield
+                             0x03, // BCunknown
+                             0x03, // BCunknown
```
I think these two entries should be removed. . .  but let me double-check that.

theresa-m replied:
I tried removing these originally but what getting an error.

a7ehuo replied:
I see these two lines were added in https://github.com/eclipse-openj9/openj9/pull/7991/files#diff-16ccd09398e392e4a6b8168ba45d9b05115abad7220fc067e37ad5df2ce9aa92 without other modifications related to these two opcode. Interesting, they couldn't be removed. Is it a build error or a test error?

hzongaro reviewed (commented):
I took a quick look at the compiler changes.  Thank you for making those changes, @theresa-m!

I had a draft pull request #18790 that had changes for allowing value type instances to be created either with `new` or with `aconst_init` and `withfield`, under the assumption that there might be a period of time where we would need to support both possibilities.  Most of the changes there are irrelevant if we're dropping those two bytecode instructions in one step.  However, I the think the change to `TR_J9ByteCodeIlGenerator::genNew` that calls `setIdentityless` and the change to `TR_EscapeAnalysis::isImmutableObject` will still be needed.  May I ask you to pick up those two changes?

I'll also have to revise OMR pull request #7239 for the change to `VPHandlers.cpp`.

hzongaro commented on runtime/compiler/ilgen/J9ByteCodeIterator.cpp:
```diff
@@ -283,10 +283,6 @@ TR_J9ByteCodeIterator::printByteCode()
       case J9BCgetstatic: case J9BCgetfield: case J9BCputstatic: case J9BCputfield:
       case J9BCcheckcast: case J9BCinstanceof:
       case J9BCnew: case J9BCanewarray:
-      case J9BCaconst_init: case J9BCwithfield:
-         printCPIndex(next2Bytes());
-         break;
-
```
I think these two lines are still needed for the other cases `J9BCgetstatic`, `J9BCgetfield`, et al.

```
printCPIndex(next2Bytes());
break;
```

theresa-m commented:
@hzongaro I added your changes from https://github.com/eclipse-openj9/openj9/pull/18790. Some of the changes were removed through merge conflicts and I also removed the `areValueTypeInstancesCreatedWithBCNew ` method. Can you please review to make sure this is what you were expecting?

hzongaro commented:
> Some of the changes were removed through merge conflicts and I also removed the `areValueTypeInstancesCreatedWithBCNew` method. Can you please review to make sure this is what you were expecting?

Yes, that's what I looking for.  Thanks for taking care of that!  (Please squash that pair of commits - we don't need to preserve `areValueTypeInstancesCreatedWithBCNew` in the history.)

theresa-m commented:
Thanks @hzongaro I squashed the commits.

a7ehuo reviewed (commented):
Thank you very much for the update @theresa-m! The JIT change looks good to me. I just have one minor comment.

hzongaro reviewed (commented):
May I ask you revise the commit message for commit e3a18a44ba667357fc5147da5e9c0d83766e9526 to remove the following paragraph, which no longer applies?

> A new `J9::ObjectModel::areValueTypeInstancesCreatedWithBCNew` method is
used to enable creation of value type instances with the new bytecode.

theresa-m commented:
> Thank you very much for the update @theresa-m! The JIT change looks good to me. I just have one minor comment.

Hey I tried removing these again and it looks like it compiles. Maybe I got confused with another spot :/ .

theresa-m commented:
> May I ask you revise the commit message for commit [e3a18a4](https://github.com/eclipse-openj9/openj9/commit/e3a18a44ba667357fc5147da5e9c0d83766e9526) to remove the following paragraph, which no longer applies?
> 
> > A new `J9::ObjectModel::areValueTypeInstancesCreatedWithBCNew` method is
> > used to enable creation of value type instances with the new bytecode.

I removed that paragraph. Is the part about using TR::newvalue still accurate?

hzongaro commented:
> Is the part about using `TR::newvalue` still accurate?

Yes - some optimizations still create value type instances with `TR::newvalue`.

hzongaro reviewed (approved):
Compiler changes look good.  Thanks!

hangshao0 commented:
Jenkins test sanity,extended zlinuxval jdknext

hangshao0 commented:
Jenkins test sanity,extended amac jdk21

hangshao0 reviewed (approved):
LGTM.
